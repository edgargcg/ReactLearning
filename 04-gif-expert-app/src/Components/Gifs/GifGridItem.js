import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({ title, url}) => {
    return (
        <div className="images__item animate__animated animate__bounce">
            <h5 className="image__title">
                {title}
            </h5>
            <img 
                src={url} 
                alt={title} 
                className="image__source" 
            />
        </div>
    )
}

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}