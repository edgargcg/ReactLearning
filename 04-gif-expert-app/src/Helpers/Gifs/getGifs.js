export const getGifs = async(category) => {
    const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(category)}&limit=5&api_key=cqDyquoPCZDl2iOHC4dDHzlf80orxefP`;
    const response = await fetch(url);
    const { data } = await response.json();
    const gifs = data.map(img => {
        return {
            id: img.id,
            title: img.title,
            url: img.images?.downsized_medium.url
        }
    });

    return gifs;
}
