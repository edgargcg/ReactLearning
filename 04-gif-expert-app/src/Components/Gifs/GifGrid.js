import React, {  } from 'react'
import { useFetchGifs } from '../../Hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
    const { data: images, loading } = useFetchGifs(category);

    return (
        <div>
            <h3>
                {category}
            </h3>
            {loading && <p>Loading</p> }
            <div className="images">
                {
                    images.map(
                        img => 
                        (
                            <GifGridItem 
                                {...img} 
                                key={img.id} 
                            />
                        )
                    )
                }
            </div>
        </div>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}