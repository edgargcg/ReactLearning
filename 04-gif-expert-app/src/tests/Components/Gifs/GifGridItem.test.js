import { shallow } from 'enzyme';
import { GifGridItem } from '../../../Components/Gifs/GifGridItem';

describe('Tests on GifGridItem', () => {
    const title = 'Batman';
    const url = 'https://holamundo.jpg';
    const wrapper = shallow(
        <GifGridItem 
            title={title} 
            url={url} 
        />
    );
    
    test('should render a GifGridItem', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should have a paragraph with title', () => {
        const paragraph = wrapper.find('h5').text().trim();
        expect(paragraph).toBe(title);
    })
    
    test('should image have all properties', () => {
        const image = wrapper.find('img');
        const alt = image.props().alt;
        const src = image.props().src;

        expect(src).toBe(url);
        expect(alt).toBe(title);
    })
    
    test('should div container have className for animation', () => {
        const div = wrapper.find('div');
        const className = div.props().className;
        
        expect(className.includes('animate__animated')).toBe(true);
    })
    

})
