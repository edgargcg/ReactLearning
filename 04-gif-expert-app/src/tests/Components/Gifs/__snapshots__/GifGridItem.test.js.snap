// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifGridItem should render a GifGridItem 1`] = `
<div
  className="images__item animate__animated animate__bounce"
>
  <h5
    className="image__title"
  >
    Batman
  </h5>
  <img
    alt="Batman"
    className="image__source"
    src="https://holamundo.jpg"
  />
</div>
`;
